src/main/kotlin/no/nav/familie/ba/sak/kjerne/beregning/TilkjentYtelseValidering.kt

```diff
@@ -42,6 +43,32 @@ fun hentBarnasAndeler(andeler: List<AndelTilkjentYtelse>, barna: List<Person>) =
  */
 object TilkjentYtelseValidering {
 
+    internal fun validerAtSatsendringKunOppdatererSatsPåEksisterendePerioder(
+        andelerFraForrigeBehandling: List<AndelTilkjentYtelse>,
+        andelerTilkjentYtelse: List<AndelTilkjentYtelse>,
+    ) {
+        val andelerGruppert = andelerTilkjentYtelse.groupBy { Pair(it.aktør, it.type) }
+        val forrigeAndelerGruppert = andelerFraForrigeBehandling.groupBy { Pair(it.aktør, it.type) }
+
+        (andelerGruppert.keys + forrigeAndelerGruppert.keys).distinct().forEach {
+            val nåværendeTidslinje = AndelTilkjentYtelseTidslinje(andelerGruppert[it] ?: emptyList())
+            val forrigeTidslinje = AndelTilkjentYtelseTidslinje(forrigeAndelerGruppert[it] ?: emptyList())
+            validerTidslinjer(forrigeTidslinje, nåværendeTidslinje)
+        }
+    }
+
+    private fun validerTidslinjer(
+        forrigeAndelerTidslinje: AndelTilkjentYtelseTidslinje,
+        nåværendeAndelerTidslinje: AndelTilkjentYtelseTidslinje,
+    ) = forrigeAndelerTidslinje.kombinerMed(nåværendeAndelerTidslinje) { forrigeAndel, nåværendeAndel ->
+        when {
+            forrigeAndel == null && nåværendeAndel != null -> throw Feil("Satsendring kan ikke legge til en andel som ikke var der i forrige behandling")
+            forrigeAndel != null && nåværendeAndel == null -> throw Feil("Satsendring kan ikke fjerne en andel som fantes i forrige behandling")
+            forrigeAndel != null && nåværendeAndel != null -> if (forrigeAndel.prosent != nåværendeAndel.prosent) throw Feil("Satsendring kan ikke endre på prosenten til en andel") else false
+            else -> false
```

Comment:
Noen lange rader her :-) Går å flytte opp prosent-sjekken, samt forenkle den med å kun sjekke not null på en av andelene. 

```suggestion
            forrigeAndel == null && nåværendeAndel != null -> 
                throw Feil("Satsendring kan ikke legge til en andel som ikke var der i forrige behandling")
            forrigeAndel != null && nåværendeAndel == null -> 
                throw Feil("Satsendring kan ikke fjerne en andel som fantes i forrige behandling")
            forrigeAndel != null && forrigeAndel.prosent != nåværendeAndel?.prosent ->
                throw Feil("Satsendring kan ikke endre på prosenten til en andel")
            else -> false
```

Reply:
Yes fikser!